src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -863,34 +869,40 @@ bool ControllerScriptEngineLegacy::evaluateScriptFile(const QFileInfo& scriptFil
 }
 
 #ifdef MIXXX_USE_QML
-std::shared_ptr<QQuickItem> ControllerScriptEngineLegacy::loadQMLFile(
+bool ControllerScriptEngineLegacy::instanciateQMLComponent(
         const LegacyControllerMapping::ScriptFileInfo& qmlScript,
-        std::shared_ptr<ControllerRenderingEngine> pScreen) {
+        QMap<QString, std::shared_ptr<ControllerRenderingEngine>>
+                availableScreens) {
+    // Like for Javascript, if the script is invalid, it should be watched so the user can fix it
+    // without having to restart Mixxx. So, add it to the watcher before
+    // evaluating it.
     VERIFY_OR_DEBUG_ASSERT(m_pJSEngine ||
             qmlScript.type !=
                     LegacyControllerMapping::ScriptFileInfo::Type::Qml) {
-        return nullptr;
+        return false;
     }
 
+    watchFilePath(qmlScript.file.absoluteFilePath());
+
     QQmlComponent qmlComponent = QQmlComponent(
             std::dynamic_pointer_cast<QQmlEngine>(m_pJSEngine).get());
 
-    QFile scene = QFile(qmlScript.file.absoluteFilePath());
-    if (!scene.exists()) {
-        qCWarning(m_logger) << "Unable to load the QML scene:" << qmlScript.file.absoluteFilePath()
+    QFile file = QFile(qmlScript.file.absoluteFilePath());
+    if (!file.exists()) {
+        qCWarning(m_logger) << "Unable to load the QML file:" << qmlScript.file.absoluteFilePath()
```

Comment:
I believe `scene` is the right terminology. I would rather we keep it instead of the cryptic `file`